nixos/tests/cockroachdb.nix

```diff
@@ -0,0 +1,24 @@
+import ./make-test.nix ({ pkgs, ...} : {
+  name = "cockroachdb";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ ];
+  };
+
+  nodes = {
```

Comment:
I think this might be overkill for `nixpkgs` CI, depending on the exact specifics of `workload`. The test I wrote is simply to boot a cluster and verify it's working at all, mainly the systemd configuration. I think sniffing for arch problems is probably better handled upstream.

Reply:
Workload testing only happens for 1 minute and takes little RAM in my tests, so I think it's fine. See #51306 for more on that.

Booting a single node is fine for testing, but generally I feel NixOS tests are already very weak in a number of areas -- I do not think a cluster test is overkill at all. Having an actual cluster test is arguably *the minimal* amount of testing we can do if we're going to offer and support a clustering database to our users, IMO! It's basically the primary feature of Cockroach. And doing an actual workload test with built-in tools is about as easy as you can ask for, I think.

I am also a strong supporter of trying to offer reasonable multi-architecture feature parity for NixOS, especially as far as packages-with-a-service-module are concerned. In other words, unless there's strong, obvious reason to reject it (e.g. significant complexity in the derivation, completely unstable and unusable) I generally aspire to offer feature-parity for services like this since they are part of NixOS proper. My indication from looking around the Cockroach upstream is that while they don't explicitly advertise `aarch64` as "production ready", it seems to work fine -- it probably just doesn't undergo the same release-stress-testing they put the x86_64 release through, I'm guessing. (And, in fact, the only reason the clustering test *doesn't* work for us on AArch64 isn't due to Cockroach, but NTP shenanigans, and that could be fixed, probably.)

Regardless, there's a lot of interacting complexity in a modern system, architecture issues aside -- and being thorough and testing things (even things upstream has control over) will save people a lot of headache later on!